Python/minestat/__init__.py

```diff
@@ -494,6 +519,161 @@ def __parse_bedrock_payload(self, payload_str: str) -> ConnStatus:
 
     return ConnStatus.SUCCESS
 
+  def fullstat_query(self) -> ConnStatus:
+    """
+    Method for querying a Minecraft Java server using the fullstat Query / GameSpot4 / UT3 protocol.
+    Needs to be enabled on the Minecraft server using:
+
+    "enable-query=true"
+
+    in the servers "server.properties" file.
+
+    This method ONLY supports full stat querys.
+    Documentation for this protocol: https://wiki.vg/Query
+    """
+    # protocol:
+    #   send handshake request
+    #   receive challenge token
+    #   send full stat request
+    #   receive status data
+
+    # Create UDP socket and set timeout
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.settimeout(self.timeout)
+
+    # padding that is prefixes to every packet
+    magic = b"\xFE\xFD"
+
+    # packettypes for the multiple packets send by the client
+    handshake_packettype = struct.pack("!B", 9)
+    stat_packettype = struct.pack("!B", 0)
+
+    # generate session id
+    session_id_int = random.randint(0, 2147483648) & 0x0F0F0F0F
+    session_id_bytes = struct.pack('>l', session_id_int)
+
+    # handshake packet:
+    #   contains 0xFE0xFD as a prefix
+    #   contains type of the packet, 9 for hanshaking in this case (encoded in Bytes as a big-endian)
+    #   contains session id (is generated randomly at the begining)
+
+    # construct the handshake packet
+    handshake_packet = magic + handshake_packettype + session_id_bytes
+
+    # send packet to server
+    sock.sendto(handshake_packet, (self.address, self.port))
+
+    try:
+      # receive the handshake response
+      handshake_res = sock.recv(24)
+
+      # extract the challenge token from the server. The beginning of the packet can be ignored.
+      challenge_token = handshake_res[5:]
+
+      # pack the challenge token into a big-endian long (int32)
+      challenge_token_bytes = struct.pack('>l', int(challenge_token[:-1]))
```

Comment:
The geyser server software seems to send two NUL bytes at the end of the handshake response? Using rstrip on the challenge_token fixes this, ignoring as many NULs as the server throws at us :D

```suggestion
      challenge_token = handshake_res[5:].rstrip(b'\00')

      # pack the challenge token into a big-endian long (int32)
      challenge_token_bytes = struct.pack('>l', int(challenge_token))
```

Error message running your branch against a geyser server:
```
  File "C:\Users\username\Programmieren\Python\minestat-demo\example.py", line 5, in <module>
    ms = minestat.MineStat('193.135.10.3', 15224, query_protocol=minestat.SlpProtocols.Query)
  File "C:\Users\username\Programmieren\Python\minestat-demo\venv\lib\site-packages\minestat\__init__.py", line 273, in __init__
    result = self.fullstat_query()
  File "C:\Users\username\Programmieren\Python\minestat-demo\venv\lib\site-packages\minestat\__init__.py", line 574, in fullstat_query
    challenge_token_bytes = struct.pack('>l', int(challenge_token[:-1]))
ValueError: invalid literal for int() with base 10: b'-599651295\x00'
```